Update react-native expo command

dev-fuad opened the pull request:
Updated the command to create react-native apps with expo to the latest one.

<img width="1552" alt="Screenshot 2024-01-22 at 10 12 07 PM" src="https://github.com/R35007/create-app-vscode-extension/assets/17686924/8cfb0c05-b661-4248-976e-8d70b055c4fa">

dev-fuad committed:
Update react-native expo command

dev-fuad committed:
fix broken parameters

R35007 commented on apps/react/react-2-native.json:
```diff
@@ -43,21 +29,10 @@
       "label": "Template",
       "description": "You can find many grea community templates by searching for <a href='https://www.npmjs.com/search?q=expo-template-*'>expo-template-*</a> on npm.",
```
Can you please change the description and point to https://github.com/expo/examples url

dev-fuad replied:
decided to keep templates instead of examples
initially I wanted to keep both but I couldn't find a way to templates/examples field conditionally or group dropdown items without making change in render logic, which I do not want to do now
and keeping both will confuse user and keeping template seems more viable than keeping examples

R35007 replied:
Sure. I have just changed the description link point to official [expo/examples](https://github.com/expo/examples)

R35007 commented on apps/react/react-2-native.json:
```diff
@@ -43,21 +29,10 @@
-      "prefix": "--template=",
+      "prefix": "-e ",
       "options": [
         { "label": "Blank" },
```
Can you please add minimum 5 templates ?

dev-fuad replied:
added all 3 templates

R35007 commented on apps/react/react-2-native.json:
```diff
@@ -89,7 +69,7 @@
       "label": "cli help",
       "description": "Click here to know more cli configurations",
-      "command": "npx expo init --help"
+      "command": "npx expo --help"
```
Please change the command to `npx create-expo-app --help`

dev-fuad replied:
done

R35007 reviewed (changes requested):
Hi @dev-fuad,

Thanks for the update. 
I have requested some changes. Can you please resolve those comments and please add the description of the change and Screen short in the PR.

Thanks,
Siva

dev-fuad committed:
updates
- add option to use npm/yarn
- use templates for templates instead of example
- update additionalCommands
fixes
- add app name pattern to avoid invalid names

R35007 committed:
feat: :art: Updated React Native App CLI options
